### DISCUSSION TITLE
Questions about the account deletion feature

### DISCUSSION THREAD

**Discussion by sunxyw** [andrewdwallo/filament-companies#32] [Q&A] [answered]
I noticed that in one commit the `is_null($user->password)` restriction to the display of the account deletion form was added, which only allows users registered through Socialite to delete their accounts, but not normal registered users .

https://github.com/andrewdwallo/filament-companies/commit/ff46a06708d720f67522e5f1fa046bf8108e5131#diff-b2a2b255af0a8e73c273dbdfb8de5b48677477c16118174c7e6b5af225e076f3R47

There is no way for normal users to set the password to `null`, does that mean they won't be able to delete the account? Is this expected behavior? I don't quite understand the reason for doing this.

**Comment by andrewdwallo** [comment, marked as answer]
No, you are right. Much of the socialite features were based on Socialstream:
https://github.com/joelbutcher/socialstream

I used their examples as an implementation for this package considering much of their features were already shown to work well with Jetstream (which is what Socialstream is connected to), and which is what this package bases its features and code from (because it is already shown to be a reputable open-source product/plugin). I understand people may not like the fact that I am using code from other sources, but I don't see any harm in doing so considering both of these sources collectively have hundreds of contributors and have been around for a good amount of time. I am currently fixing many of the issues that these packages have in a branch I am working on... which will be released in the next couple of days hopefully. Many things may change so I would make sure that you save anything you may be working on (if you really want to keep it), so that you can implement it in the upcoming release... because there may be breaking changes, and you may have to redownload. I am sorry that this wasn't fixed beforehand, but at least it is better fix anything now then later down the road when many more users could potentially be using this plugin... But anyways, thank you for bringing this up!

**Reply by sunxyw** [reply]
Thank you very much for your response. I don't think there's anything wrong with borrowing someone else's code if the license allows it, which is, in my opinion, what open source is all about.
I'm looking forward to the next update, which I hope will bring even more improvements and features.
Thank you once more for keeping such an excellent project; it definitely aids my project development.

**Reply by andrewdwallo** [reply]
Yeah of course. You are also always welcome to suggest features that would help as well, couldn't really help you with what you may need if I don't know... I am always open to suggestions. The discord channel for my plugin is here: https://discord.com/channels/883083792112300104/1059008724410310767
